dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/GithubAppCheck.java

```diff
@@ -0,0 +1,134 @@
+     * @param fullRepoName = The repository full name, i.e, of the format "owner/repoName". Eg: "Salesforce/dockerfile-image-update"
+     * @return True if github app is installed, false otherwise. 
+     */
+    protected boolean isGithubAppEnabledOnRepository(String fullRepoName){
+        refreshJwtIfNeeded(appId, privateKeyPath);
+        try {
+            gitHub.getApp().getInstallationByRepository(fullRepoName.split("/")[0], fullRepoName.split("/")[1]);
+            return true;
+        } catch (HttpException exception) {
+            if (exception.getResponseCode() != 404) {
+                // Log for any HTTP status code other than 404 Not found. 
+                log.warn("Caught a HTTPException {} while trying to get app installation. Defaulting to False", exception.getMessage());
+            }
+            return false;
+        } catch (IOException exception) {
+            // Most often happens on timeout scenarios. 
+            log.warn("Caught a IOException {} while trying to get app installation. Defaulting to False", exception.getMessage());
+            return false;
+        }
+    }
+
+    /**
+     * Method to refresh the JWT token if needed. Checks the JWT expiry time, and if it is 60s away from expiring, refreshes it. 
+     * @param appId = The id of the Github App to generate the JWT for
+     * @param privateKeyPath = The path to the private key of the Github App to generate the JWT for
+     */
+    private void refreshJwtIfNeeded(String appId, String privateKeyPath){
+        if (jwt == null || jwtExpiry.isBefore(Instant.now().minusSeconds(60))) {  // Adding a buffer to ensure token validity
+            try {
+                generateJWT(appId, privateKeyPath);
+            } catch (IOException | GeneralSecurityException exception) {
+                log.warn("Could not refresh the JWT due to exception: {}", exception.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Method to generate the JWT used to access the Github App APIs. We generate the JWT to be valid for 600 seconds. 
+     * Along with the JWT value, the jwtExpiry value is set to the time of 600 sec from now. 
+     * @param appId = The id of the Github App to generate the JWT for
+     * @param privateKeyPath = The path to the private key of the Github App to generate the JWT for
+     * @throws IOException
+     * @throws GeneralSecurityException
+     */
+    private void generateJWT(String appId, String privateKeyPath) throws IOException, GeneralSecurityException {
+        Security.addProvider(new org.bouncycastle.jce.provider.BouncyCastleProvider());
+        RSAPrivateKey privateKey = getRSAPrivateKey(privateKeyPath);
+
+        Algorithm algorithm = Algorithm.RSA256(null, privateKey);
+        Instant now = Instant.now();
+        jwt = JWT.create()
+                .withIssuer(appId)
+                .withIssuedAt(Date.from(now))
+                .withExpiresAt(Date.from(now.plusSeconds(600))) // 10 minutes expiration
```

Comment:
why is this hardcoded?

Reply:
what will the behavior be if you set this really short (for example 1 second)

Reply:
@afalko we would like to avoid as much API calls as possible. Setting to 1 second means that we have to make an API call to regenerate a new JWT token every 1 second. Taking into account that the DFIU takes hours to run, I believe it is not efficient